### DISCUSSION TITLE
"Not the next claimed resources" error after upgrade to v5.6

### DISCUSSION THREAD

**Discussion by jfaigan** [openTCS/opentcs#56] [Q&A] [answered]
I am getting the following error when trying to execute a TO in my model.

![image](https://user-images.githubusercontent.com/14929031/197019235-eb1c7709-eb8d-4752-b33b-9dd1fecad4ad.png)

This model was working in v5.2.

I did not see any reference to breaking changes in the release notes.

Any idea what might be generating this error?

Joel

**Comment by swltr** [comment]
I am not convinced this is related to the plant model content, yet. The following information could help with identifying the cause:

* Are you testing this with the loopback driver / virtual vehicle, or are you using some other (custom) vehicle driver?
* Can you reproduce it with the loopback driver and/or the demo model that openTCS comes with?
* How are you setting the system up before creating transport orders? Which steps do you take for making the vehicle available for transport orders? (I.e. how exactly does the vehicle get to its initial position and state?)
* In https://github.com/openTCS/opentcs/discussions/50#discussioncomment-3926509, you mentioned that you are running it on Java 17. Currently, we are still developing using Java 13 and thus haven't tested with more recent versions. I would not expect this to have any influence here, but maybe you could check with 13, just to be sure.

**Reply by jfaigan** [reply]
Ok, here are some answers to your questions...

- Yes. I am testing with the loopback driver.  I do not have access to any real vehicles.
- No, I am unable to reproduce the error with the Demo model.  I was able to send a TO and have the vehicle move between locations.
- Setup: I am using the Kernel Control Center to activate and assign an initial position to a vehicle.  I then enable to the vehicle to accept commands in the Operations Desk.
![image](https://user-images.githubusercontent.com/14929031/197197394-e8da1cd0-f3e6-4563-bf8a-34a5d8392249.png)
![image](https://user-images.githubusercontent.com/14929031/197197508-3d41f45e-8160-4c3a-80e3-40712d1abe3c.png)

- As the Demo model runs with v17, I do not believe this is the issue.  Also, v13 does not seem to be available anymore.
![image](https://user-images.githubusercontent.com/14929031/197197755-e73b0681-f2f2-4fc4-bb3e-5d414c092b62.png)

I tried again with my model after being successful with the Demo model and I am still having the error, however, I have found the error.

When I loaded the demo model I noticed that all the points between the locations were set to be Halt Points.  I had some Report Points between my locations and this is where it was failing.  

In this image you can see that Point-0004 is a Report Point and it fails.  If I set the type to be Halt point, it will work like it should.
![image](https://user-images.githubusercontent.com/14929031/197200366-e6ae4d5c-b842-488e-8b35-2675bfbaa770.png)


I do not know if that is a bug or just a lack of my understanding.

Please let me know if you need more info on this error.

Joel

**Reply by swltr** [reply]
Ah. But the vehicle did not start from a report point, and its destination wasn't a report point, either? (Report points are intended to (only) be intermediate points that some vehicle types can report while moving from one halt point to the next.)

**Reply by jfaigan** [reply]
Yes, the vehicle began on a Halt Point and was destined for another Halt Point.  There is a Report Point between them.  If I remove the Report Point then I do not get the error.

This is my original model.  Point-0002 and 0004 are Report Points. All others are Halt Points.
![image](https://user-images.githubusercontent.com/14929031/197524677-315066b6-5a48-4adc-b92a-e3e7d0513977.png)

**Reply by jfaigan** [reply]
This is what was sent to the server:
![image](https://user-images.githubusercontent.com/14929031/197524963-538e382d-b0ab-48d8-adc5-cbc2d4c185d2.png)

The Vehicle begins at Point-001

**Reply by jfaigan** [reply]
This is what was received from the REST service:
![image](https://user-images.githubusercontent.com/14929031/197525208-f0e2a942-9f5e-4680-be3c-21cc89688058.png)

**Reply by swltr** [reply]
I can confirm this behaviour and it is definitely not intended; I have opened #59 for this bug. Thank you for reporting it!

In the meantime, the obvious workaround is to not use report points in the plant model. (Report points are a feature that is rarely used these days. If you have simply been trying it out, you're probably fine without it. If you have a use case that really makes use of report points, it would be nice to hear about it, as this feature is now and then discussed to be obsolete and a candidate for removal.)

**Reply by jfaigan** [reply]
Ok!  Glad I could help!

With regards to the use case, as you mentioned, at this point I am just trying it out so there is no real use case in mind.  My thought process behind the setup was that, in my mind, it made sense to only have halt points at locations and everything in between would just be for reporting status.  If that is not how the industry works, that is good to know.  I am looking at this more as an integrator...sending information to the fleet manager based on operations on the shop floor.

Keep up the great work.  I really like the software.  It is very easy to use and setup.

**Comment by jfaigan** [comment, marked as answer]
An issue has been opened for this item. 
https://github.com/openTCS/opentcs/issues/59
